### DISCUSSION TITLE
How to configure trivy operator with helm chart with a different path for the containerd socket.

### DISCUSSION THREAD

**Discussion by sarg3nt** [aquasecurity/trivy-operator#2352] [Help & Support]
Our OS stores the containerd socket in `/run/k3s/containerd/containerd.sock` and I can't figure out how to tell Trivy where the socket is with the Helm chart.
I've looked here and found one other question where someone ran into the same thing but they didn't actually say how they fixed it.

I see the [docs](https://trivy.dev/v0.57/docs/references/troubleshooting/#containerd) that say we can set `CONTAINERD_ADDRESS` but there does not appear to be a way to set `env` vars in the [Helm](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/values.yaml) chart.

Here's the error we are getting.

```
message":"2024-12-06T17: 33: 22Z     FATAL     Fatal error     image scan error: scan error: unable to initialize a scanner: unable to initialize an image scanner: unable to find the specified image "memcached:1.6.23-alpine" in [    "docker""containerd""podman""remote"
  ]: 4 errors occurred:
     * docker error: unable to inspect the image (memcached: 1.6.23-alpine): Cannot connect to the Docker daemon at unix: ///var/run/docker.sock. Is the docke
r daemon running?
     * containerd error: containerd socket not found: /run/containerd/containerd.sock
     * podman error: unable to initialize Podman client: no podman socket 
found: stat podman/podman.sock: no such file or directory
     * remote error: GET https: //index.docker.io/v2/library/memcached/manifests/sha256:56b3526ad7eba067730ca64ab7a9062
  327160c9ff6dd9bc6d84fe0cd36b9005b: TOOMANYREQUESTS: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https: //www.docker.com/
increase-rate-limit
```

**Comment by afdesk** [comment]
@sarg3nt 
 there is another known issue in your logs -  `TOOMANYREQUESTS`.
using another registry (ex. mirror.gcr.io) should help with this error. 

about other question.
I'd try to set up evns here: https://github.com/aquasecurity/trivy-operator/blob/023fedff4f66f8457b4f0be9e99bffb3c0e7cdbc/deploy/helm/values.yaml#L601-L605

**Comment by belaw** [comment]
I have the same issue. **Trivy operator + private registry can't be used without workarounds on a k3s cluster**. Probably affects RKE2 too.

In k3s, private container registries are configured on containerd - https://docs.k3s.io/installation/private-registry
That means that there are no image pull secrets required anywhere for the configured private registries.
In k3s, the containerd socket is located at a non standard path `/run/k3s/containerd/containerd.sock` - https://docs.rke2.io/reference/cli_tools#containerd

It's impossible to configure trivy operator to use a global image pull secret, only per namespace.
It's impossible to configure the scan jobs to use a custom containerd socket.
The only solution is to use the filesystem workaround. (https://aquasecurity.github.io/trivy-operator/latest/tutorials/private-registries/#first-option-filesystem-scanning)

This should really be fixed

**Reply by afdesk** [reply]
@belaw thanks for the report.
we should create a new issue to track it, right?

**Reply by afdesk** [reply]
to track #2586

**Comment by afdesk** [comment]
After v0.27 there is a way to configure scan jobs with Trivy config file.
so it should resolve this issue.
